frontend/src/component/releases/ReleasePlanTemplate/MilestoneList.tsx

```diff
@@ -0,0 +1,76 @@
+import type { IReleasePlanMilestonePayload } from 'interfaces/releasePlans';
+import { useState } from 'react';
+import { MilestoneCard } from './MilestoneCard';
+import { styled } from '@mui/material';
+import { Button } from '@mui/material';
+import Add from '@mui/icons-material/Add';
+
+interface IMilestoneList {
+    milestones: IReleasePlanMilestonePayload[];
+    setMilestones: React.Dispatch<
+        React.SetStateAction<IReleasePlanMilestonePayload[]>
+    >;
+    setAddStrategyOpen: (open: boolean) => void;
+    errors: { [key: string]: string };
+    clearErrors: () => void;
+}
+
+const StyledAddMilestoneButton = styled(Button)(({ theme }) => ({
+    marginTop: theme.spacing(1),
+    maxWidth: theme.spacing(20),
+}));
+
+export const MilestoneList = ({
+    milestones,
+    setMilestones,
+    setAddStrategyOpen,
+    errors,
+    clearErrors,
+}: IMilestoneList) => {
+    const [currentMilestone, setCurrentMilestone] = useState(-1);
```

Comment:
Why not use `IReleasePlanMilestonePayload | undefined` here? Should be more explicit than `-1` and indexes.

Reply:
This one is for adding strategies to the milestone. Current milestone (should probably be named ...index) gets passed to the add strategy form. If we're adding something to that object (like we do with names) it should be bubbled up no?